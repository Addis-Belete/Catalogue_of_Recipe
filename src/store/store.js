import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import { recipeReducer } from '../reducer';

const store = createStore(recipeReducer, composeWithDevTools(applyMiddleware(thunk)));
export default store;
