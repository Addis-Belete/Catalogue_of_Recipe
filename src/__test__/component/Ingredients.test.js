import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Ingredients from '../../container/Ingeredients';
import store from '../../store/store';

afterEach(cleanup);
it('displays Ingredients', () => {
  const recipe = [{ title: 'add', id: 123, image: '.jepg' }];
  render(
    <Provider store={store}>
      <BrowserRouter>
        <Ingredients recipes={recipe} />
      </BrowserRouter>
    </Provider>,
  );
  expect(screen.getByText('Ingredients')).toBeTruthy();
});

it('should render the list of ingredients', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Ingredients />
      </BrowserRouter>
    </Provider>,
  );
  expect(
    asFragment(
      <Provider store={store}>
        <Ingredients />
      </Provider>,
    ),
  ).toMatchSnapshot();
});
