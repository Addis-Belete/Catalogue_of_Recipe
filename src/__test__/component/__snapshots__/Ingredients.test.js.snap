// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the list of ingredients 1`] = `
<DocumentFragment>
  <div
    class="close-ingredients ingredient tablet"
    role="button"
    tabindex="0"
  >
    <div
      class="ingredient-list"
    >
      <div>
        <span
          class="close-ingredients span"
          role="button"
          tabindex="0"
        >
          Close
        </span>
      </div>
      <h2
        class="h2"
      >
        Ingredients
      </h2>
      <ul
        class="ul"
      >
        <li
          class="li"
        >
          egg white
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
